### PR TITLE
Jon/aave/integrate-loan-manager

### PR DESCRIPTION
**Pull request by Jon-edge**
This PR integrates LoanManager into the AAVE GUI scenes and user workflow.  

Routing has also been updated such that from the LoanStatusScene of a newly created loan: 
- The back button will transition to the LoanDashboardScene
- A new LoanSummaryCard will be shown representing the newly created loan
- Clicking on this new LoanSummaryCard will transition to the LoanDetailsScene where the status of the ActionProgram may be viewed in a card*
- Clicking on the ActionProgram status card will transition back to the LoanStatusScene

This PR also includes fixes:
- setText of undefined error
- Missing fee tile fix
- APR localization and precision fix on LoanDetailsScene

KNOWN ISSUES:
- Any ActionProgram consisting of a single step will not display correctly in the preliminary ActionProgram status card*, as an update to the display mechanisms of the ActionQueue is required.

*A preliminary design was added to assist in testing purposes and serve as a general starting point for the design of the ActionQueue status display:

#### PR Dependencies

https://github.com/EdgeApp/edge-react-gui/pull/3605

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202895674767132
  - https://app.asana.com/0/0/1202827562867770
  - https://app.asana.com/0/0/1202895674767143

### CODE REVIEW

**Review by samholmes — CHANGES_REQUESTED** [review]
This is great work!! 👍 👍 

Granted it's a large PR, so changes are needed.

File: src/components/Main.ui.js [hunk 1]
```diff
@@ -967,6 +967,10 @@ export class MainComponent extends React.Component<Props> {
         return true
       }
     }
+    if (this.isCurrentScene('loanCreateStatus')) {
+      Actions.jump('loanDashboard')
+      return true
+    }
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/Main.ui.js:981]: The issue is that the back button cannot be controlled by the scene itself, it's handled here.

File: src/components/Main.ui.js [hunk 2]
```diff
@@ -909,7 +909,15 @@ export class MainComponent extends React.Component<Props> {
...
-              key="loanStatus"
+              key="loanCreateStatus"
+              component={withNavigation(ifLoggedIn(LoanStatusScene))}
+              navTransparent
+              renderLeftButton={<BackButton onPress={this.handleBack} />}
+              renderRightButton={<SideMenuButton />}
+              onLeft={Actions.pop}
+            />
+            <Scene
+              key="loanDetailsStatus"
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/Main.ui.js:920]: To give Main.ui information about where the transition into LoanStatus came from, and to allow it to control where the back button should route.

File: src/components/scenes/Loans/LoanCreateConfirmationScene.js [hunk 3]
```diff
@@ -98,9 +100,10 @@ export const LoanCreateConfirmationScene = (props: Props) => {
   }, [borrowApprovalAction, depositApprovalAction, srcWallet])
 
   const onSliderComplete = async (resetSlider: () => void) => {
-    if (actionProgram != null) {
+    if (actionProgram != null && loanAccount != null) {
       try {
-        await dispatch(scheduleActionProgram(actionProgram))
+        await dispatch(createLoanAccountStore(loanAccount))
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanCreateConfirmationScene.js:110]: createLoanAccountStore already does check if the LA exists and will throw if it does.

Yes, 
1. We SHOULD never get to this block more than once, 
2. There should have been no way for createLoanAccountStore to fail, for the reason outlined above or any other reason 

Therefore, I am asserting these assumptions.

[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanCreateConfirmationScene.js:110]: If the loan account exists already in the store, something is wrong. The assumption is that this scene is responsible for creating a unique new loan account, and yes, we should never get to this scene if the user attempts to re-use a base wallet for a new loan account.

Those assumptions are wrapped in this assertion.

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 4]
```diff
@@ -73,9 +73,9 @@ export const LoanDashboardScene = (props: Props) => {
   const [isNewLoanLoading, setIsNewLoanLoading] = useState(false)
 
   // TODO: When new loan dApps are added, we will need a way to specify a way to select which dApp to add a new loan for.
+  const hardPluginWalletIds = Object.keys(wallets).filter(walletId => wallets[walletId].currencyInfo.pluginId === hardWalletPluginId)
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDashboardScene.js:76]: Good catch, I agree

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 5]
```diff
@@ -77,16 +77,13 @@ export const LoanDashboardScene = (props: Props) => {
...
-            newLoanWallet = wallets[walletId]
-          }
-        }
-      )
-    else if (hardPluginWalletIds.length === 1) {
+      const { walletId: newWalletId } = await Airship.show(bridge => (
+        <WalletListModal bridge={bridge} headerTitle={s.strings.select_wallet} allowedAssets={[{ pluginId: hardWalletPluginId }]} />
+      ))
+      newLoanWallet = newWalletId != null ? wallets[newWalletId] : null
+    } else if (hardPluginWalletIds.length === 1) {
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDashboardScene.js:86]: Changed this airship call to an await. Seemed using .then inside an async handler wasn't working

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 6]
```diff
@@ -128,9 +125,11 @@ export const LoanDashboardScene = (props: Props) => {
...
   const footer = isNewLoanLoading ? (
     // Render a loading card in place of the "New Loan" button while initializing a new loan
-    <Card marginRem={[0, 0.5, 0, 0.5, 0]}>
-      <FillLoader />
-    </Card>
+    <>
+      <Card marginRem={[0, 0.5, 0, 0.5, 0]}>
+        <FillLoader />
+      </Card>
+    </>
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDashboardScene.js:132]: Commit message wasn't updated, should just be "Update LoanDashboardScene to show newly created LoanAccounts, Fix Wallet Picker not properly selecting a wallet"

Originally I was setting this scene up to handle showing multiple new LoanAccounts from the end of the LoanCreate workflow (different from clicking "New Loan" here).
Instead of designing some new card or LoanSummaryCard mutation based on yet another condition, I later changed it to simply reuse the existing LoanSummaryCard as-is for new LoanAcccounts.

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 7]
```diff
@@ -167,12 +152,12 @@ export const LoanDashboardScene = (props: Props) => {
...
       <EdgeText style={styles.textSectionHeader}>{s.strings.loan_active_loans_title}</EdgeText>
       <FlatList
-        data={borrowInfos}
+        data={Object.keys(loanAccounts)}
         keyboardShouldPersistTaps="handled"
         renderItem={renderLoanCard}
         style={margin}
         ListFooterComponent={footer}
-        keyExtractor={(borrowInfo: TempBorrowInfo) => borrowInfo.borrowEngine.currencyWallet.id ?? '0'}
+        keyExtractor={(loanAccount: LoanAccount) => (loanAccount != null ? loanAccount.id : '0')}
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDashboardScene.js:161]: Oops, you're right

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 8]
```diff
@@ -145,15 +138,7 @@ export const LoanDashboardScene = (props: Props) => {
...
-  if (borrowInfosError != null)
-    return (
-      <SceneWrapper background="theme" hasTabs={false}>
-        <SceneHeader underline title={s.strings.loan_dashboard_title} />
-        <Alert title={s.strings.loan_error_title} type="error" message={translateError(borrowInfosError)} />
-      </SceneWrapper>
-    )
-
-  if (borrowInfos == null) {
+  if (loanAccounts == null || !isWalletsLoaded) {
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDashboardScene.js:142]: Yes, temporarily. They eventually load but it takes time. It could be related to the issue RJ was observing where an account with many wallets takes some time to load...

File: src/components/scenes/Loans/LoanDetailsScene.js [hunk 9]
```diff
@@ -85,15 +111,32 @@ export const LoanDetailsScene = (props: Props) => {
...
+  }
+
   return (
     <SceneWrapper>
-      <SceneHeader underline title={s.strings.loan_details_title} style={styles.sceneHeader}>
-        <Space right>
-          <Ionicon name="information-circle-outline" size={theme.rem(1.25)} color={theme.iconTappable} />
-        </Space>
-      </SceneHeader>
+      <SceneHeader underline title={s.strings.loan_details_title} style={styles.sceneHeader} />
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDetailsScene.js:138]: Could you create a task for what you mean? I don't see any icons in any of the scene headers in the design spec...

[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanDetailsScene.js:138]: It's not just the SceneHeaders that can have information icons. Maybe we just need a generic info button component that hides itself automatically if no link is given. I'm sure this won't be the last time we need an info button for a copy we don't have yet.

https://app.asana.com/0/1200868386910353/1202895674767137/f

File: src/components/scenes/Loans/LoanBorrowDetailsScene.js [hunk 10]
```diff
@@ -34,11 +34,12 @@ export const LoanBorrowDetailsScene = (props: Props) => {
...
   const { route, navigation } = props
   const { params } = route
-  const { borrowPlugin } = params
+  const { loanAccount } = route.params
+  const { borrowEngine: initBorrowEngine, borrowPlugin } = loanAccount
 
-  // Refreshing borrowEngine
-  const borrowEngineRefresher = useCallback(() => borrowPlugin.makeBorrowEngine(params.borrowEngine.currencyWallet), [borrowPlugin, params.borrowEngine])
-  const borrowEngine = useRefresher<BorrowEngine>(borrowEngineRefresher, params.borrowEngine, 10000)
+  // Refreshing borrowEngine TODO: refactor common method
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanBorrowDetailsScene.js:44]: Created a breakaway task, can just consolidate once it's actually handled properly:

https://app.asana.com/0/1200382638405084/1202895685301473/f

File: src/components/scenes/Loans/LoanCreateConfirmationScene.js [hunk 11]
```diff
@@ -120,7 +121,9 @@ export const LoanCreateConfirmationScene = (props: Props) => {
 
   if (loanAccountError != null) return <Alert title={s.strings.error_unexpected_title} type="error" message={translateError(loanAccountError)} />
 
-  return (
+  return loanAccount == null ? (
+    <FillLoader />
+  ) : (
```
Comment (samholmes) [inline src/components/scenes/Loans/LoanCreateConfirmationScene.js:126]: This may not belong in `fixup! Update AAVE scene route props to use LoanManager, update routing`. Either it applies to another commit, or stands on its own?

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 12]
```diff
@@ -30,6 +30,7 @@ import { SceneHeader } from '../../themed/SceneHeader'
 type Props = {
   navigation: NavigationProp<'loanDashboard'>
 }
+const HARD_WALLET_PLUGIN_ID = 'polygon'
```
Comment (samholmes) [inline src/components/scenes/Loans/LoanDashboardScene.js:33]: This change also deserves its own commit.

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 13]
```diff
@@ -112,34 +111,34 @@ export const LoanDashboardScene = (props: Props) => {
     }
   })
 
-  const renderLoanCard = useHandler((item: FlatListItem<string>) => {
-    const loanAccount: LoanAccount = loanAccounts[item.item]
+  const renderLoanCard = useHandler((item: FlatListItem<LoanAccount>) => {
```
Comment (samholmes) [inline src/components/scenes/Loans/LoanDashboardScene.js:114]: This type change is a fixup for 90cadded413a2fa024352409bb3b28731e1ae0f0

File: src/components/scenes/Loans/LoanDashboardScene.js [hunk 14]
```diff
@@ -112,34 +111,34 @@ export const LoanDashboardScene = (props: Props) => {
...
     const handleLoanPress = () => {
-      navigation.navigate('loanDetails', { loanAccount })
+      navigation.navigate('loanDetails', { loanAccountId: loanAccount.id })
     }
     return (
       <LoanSummaryCard onPress={handleLoanPress} borrowEngine={loanAccount.borrowEngine} iconUri={iconUri} exchangeRates={exchangeRates} key={loanAccount.id} />
     )
   })
 
+  const isCompatibleWalletsAvailable =
```
Comment (samholmes) [inline src/components/scenes/Loans/LoanDashboardScene.js:125]: This was meant for 275ab117a7b1c3e6cef334a37d6c4d77dc7af8ea

Comment (samholmes) [reply src/components/scenes/Loans/LoanDashboardScene.js:125]: I'll save commenting on any other fixup misplacements because I'm sure this will all be resolved in the rebase.

File: src/components/scenes/Loans/LoanStatusScene.js [hunk 15]
```diff
@@ -27,8 +27,8 @@ import { MainButton } from '../../themed/MainButton'
 import { SceneHeader } from '../../themed/SceneHeader'
 
 type Props = {
-  navigation: NavigationProp<'loanDetailsStatus'>,
-  route: RouteProp<'loanDetailsStatus'>
+  navigation: NavigationProp<'loanStatus'>,
+  route: RouteProp<'loanStatus'>
```
Comment (samholmes) [inline src/components/scenes/Loans/LoanStatusScene.js:31]: Is this an accidental revert?

File: src/components/scenes/Loans/ManageCollateralScene.js [hunk 16]
```diff
@@ -198,9 +202,10 @@ export const ManageCollateralScene = <T: $Keys<ParamList>>(props: Props<T>) => {
 
   const onSliderComplete = async (resetSlider: () => void) => {
     if (actionOp != null) {
-      const actionProgram = await makeActionProgram(actionOp, actionOpType)
+      const actionProgram = await makeActionProgram(actionOp)
       try {
         await dispatch(scheduleActionProgram(actionProgram))
+        await dispatch(runLoanActionProgram(loanAccount, actionProgram, actionOpType))
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/components/scenes/Loans/ManageCollateralScene.js]: This remains unresolved

File: src/components/scenes/Loans/LoanStatusScene.js [hunk 17]
```diff
@@ -27,8 +27,8 @@ import { MainButton } from '../../themed/MainButton'
 import { SceneHeader } from '../../themed/SceneHeader'
 
 type Props = {
-  navigation: NavigationProp<'loanStatus'>,
-  route: RouteProp<'loanStatus'>
+  navigation: NavigationProp<'loanDetailsStatus'>,
+  route: RouteProp<'loanDetailsStatus'>
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/components/scenes/Loans/LoanStatusScene.js]: Ah, looks like you did revert this but partially.. Missing changes in routerTypes?

**Review by samholmes — APPROVED** [review]
Approved after fixups applied (you may need to rework some fixup commits before before they can be applied). Also take not on the partial revert on the `loanStatus` scene key.

**Review by samholmes — APPROVED** [review]
